targets/unity-v2/Testing/Tests/PersistentSockets/PubSubTest.cs

```diff
@@ -0,0 +1,103 @@
+            // this test will ping a write event continuously
+            // THIS TEST THROTTLES THE TITLE. so you should NOT do this until you KNOW the object is open
+            if (pubSub != null && pubSub.State == PersistentSocketState.Opened && testTicks % playstreamTickDelay == 0)
+            {
+                EventsModels.WriteEventsRequest req = new EventsModels.WriteEventsRequest();
+
+                EventsModels.EventContents ec = new EventsModels.EventContents();
+
+                ec.Entity = new EventsModels.EntityKey();
+                ec.Entity.Id = _MyEntityKey.Id;
+                ec.Entity.Type = _MyEntityKey.Type;
+                ec.Name = "testevent";
+
+                ec.EventNamespace = "com.playfab.events.test";
+
+                req.Events = new List<EventsModels.EventContents>();
+                req.Events.Add(ec);
+
+                PlayFabEventsAPI.WriteEvents(req, WriteEventSuccessful, WriteEventFail);
+            }
+        }
+
+        public void WriteEventSuccessful(EventsModels.WriteEventsResponse response)
+        {
+            // check that we have assigned id's?
+            // this will probably get called a few times before the hub connection opens
+            // but once the hub connection opens it should immediatley start seeing something each Tick
+        }
+
+        public void WriteEventFail(PlayFabError error)
+        {
+            Debug.Log("Write Event Failed: " + error.ErrorMessage);
+        }
+
+        public override void TearDown(UUnitTestContext testContext)
+        {
+            PlayFabSettings.TitleId = _previousTitleId;
+            PlayFabSettings.VerticalName = null;
+        }
+
+        public override void ClassTearDown()
+        {
+        }
+
+        [UUnitTest]
+        public void TestPubSubConstruction(UUnitTestContext testContext)
+        {
+            ClientModels.LoginWithCustomIDRequest login = new ClientModels.LoginWithCustomIDRequest();
+            login.CustomId = "PersistentSocketsUnityUnitTest" + Guid.NewGuid().ToString();
+            login.CreateAccount = true;
+            PlayFabClientAPI.LoginWithCustomID(login, LoginSuccess, LoginFailure, testContext);
+        }
+
+        void LoginSuccess(ClientModels.LoginResult result)
+        {
+            _MyEntityKey = new PlayFab.EventsModels.EntityKey { Id = result.EntityToken.Entity.Id, Type = result.EntityToken.Entity.Type };
+
+            pubSub = new PubSub(message =>
+            {
+                ((UUnitTestContext)result.CustomData).EndTest(UUnitFinishState.PASSED, "PubSub construction successful");
+            },
+            new Topic { EventNamespace = "com.playfab.events.test", Name = "testevent", Entity = new Entity { Type = _MyEntityKey.Type, Id = _MyEntityKey.Id } });
+        }
+
+        void LoginFailure(PlayFab.PlayFabError error)
+        {
+            ((UUnitTestContext)error.CustomData).Fail("PubSub UnitTest Login Failed with the message: " + error.ErrorMessage);
```

Comment:
There is a more thorough string that can be accessed: error.GenerateErrorReport().
ErrorMessage is only one of potentially many relevant strings contained within an error obj.